src/FlowDiagnostics.jl

```diff
@@ -486,56 +486,89 @@ function (MLD::MixedLayerDepth)(i, j, k, grid, args...)
+
+    k_val  = anomaly(criterion, i, j, k, grid, args...)
+    k⁺_val = anomaly(criterion, i, j, k + 1, grid, args...)
+
+    zₖ = znode(i, j, k, grid, Center(), Center(), Center())
+    z₊ = znode(i, j, k+1, grid, Center(), Center(), Center())
+
+    return zₖ + (z₊ - zₖ) * (ref + δ - k_val) / (k⁺_val - k_val)
+end
+
 """
     $(SIGNATURES)
 
 Defines the mixed layer to be the depth at which the density is more than `pertubation`
 greater than the surface density.
 
 When this model is used, the arguments `buoyancy` and `C` should be 
-supploed where `buoyancy` should be the buoyancy model, and `C` should be a named 
+supplied where `buoyancy` should be the buoyancy model, and `C` should be a named 
 tuple of `(; T, S)`, `(; T)` or `(; S)` (the latter two if the buoyancy model 
 specifies a constant salinity or temperature).
 """
-struct DensityAnomalyCriterion{FT}
+struct DensityAnomalyCriterion{FT} <: AbstractAnomalyCriterion
     pertubation :: FT
 end
 
 validate_criterion_model(::DensityAnomalyCriterion, args...) = 
     @error "For DensityAnomalyCriterion you must supply the arguments buoyancy and C, where C is a named tuple of (; T, S), (; T) or (; S)"
 
-validate_criterion_model(::DensityAnomalyCriterion, buoyancy, C) = 
-    @error "DensityAnomalyCriterion is not implemented for your buoyancy, model as the density anomaly is not defined"
+validate_criterion_model(::DensityAnomalyCriterion, buoyancy, C) = nothing
     
 validate_criterion_model(::DensityAnomalyCriterion, ::SeawaterBuoyancy{<:Any, <:BoussinesqEquationOfState}, C::NamedTuple) = nothing
 
-@inline function density(i, j, k, grid, b::SeawaterBuoyancy{<:Any, <:BoussinesqEquationOfState}, C)
+@inline function anomaly(::DensityAnomalyCriterion, i, j, k, grid, b::SeawaterBuoyancy{<:Any, <:BoussinesqEquationOfState}, C)
     T, S = get_temperature_and_salinity(b, C)
 
     return ρ′(i, j, k, grid, b.equation_of_state, T, S)
 end
 
-@inline function (density_criterion::DensityAnomalyCriterion)(i, j, k, grid, buoyancy, C)
-    δρ = density_criterion.pertubation
-
-    ρᵣ = (density(i, j, grid.Nz, grid, buoyancy, C) + density(i, j, grid.Nz+1, grid, buoyancy, C)) * convert(eltype(grid), 0.5)
+"""
+    $(SIGNATURES)
 
-    ρₖ = density(i, j, k, grid, buoyancy, C)
+Defines the mixed layer to be the depth at which the buoyancy is more than `pertubation`
+greater than the surface buoyancy (but the pertubaton is usually negative).
 
-    return ρₖ > ρᵣ + δρ
+When this model is used, the arguments `buoyancy` and `C` should be 
+supplied where `buoyancy` should be the buoyancy model, and `C` should be a named 
+tuple of `(; T, S)`, `(; T)` or `(; S)` (the latter two if the buoyancy model 
+specifies a constant salinity or temperature).
+"""
+@kwdef struct BuoyancyAnomalyCriterion{FT} <: AbstractAnomalyCriterion
+    pertubation :: FT = - 1/8 * 1020 / g_Earth
```

Comment:
Something is off here. Buoyancy has units of gravitational acceleration (not 1/g). So it makes sense to express the buoyancy anomaly as `c * g`, where `c` is some non-dimensional number that represents your interpretation of a reasonable mixed layer depth for Boussinesq.

Reply:
Oh I put this the wrong way around it should be - g * 0.125 / 1020

Reply:
How about just `- 1e-4 * g` as the default?